python_ta/cfg/graph.py

```diff
@@ -2,8 +2,11 @@
 
 from typing import Any, Generator, List, Optional, Set, Tuple, Union
 
+import z3
```

Comment:
Because `z3` is an _optional_ dependency, we should be careful with how this is being imported. One strategy is to use a try-except:

```python
try:
    import z3
except ImportError:
    z3 = None
```

And then check whether `z3 is None` below.

Reply:
Assuming `z3 is None`, what is to be done? 

Is most of the file's body meant to be nested under an if-statement verifying that `z3 is not None`?

Reply:
> Because `z3` is an _optional_ dependency, we should be careful with how this is being imported. One strategy is to use a try-except:
> 
> ```python
> try:
>     import z3
> except ImportError:
>     z3 = None
> ```
> 
> And then check whether `z3 is None` below.

I think I am overcomplicating something because this piece of advice led me down a path to many different points of failure. I don't believe it is related to why the tests are failing, as I have isolated the imports and nothing changed. Please advise as to whether I have taken the right approach.